app/main/views/users.py

```diff
@@ -1,16 +1,28 @@
 from datetime import datetime
-from dmapiclient.audit import AuditTypes
+
+from flask import abort, current_app, jsonify, request
+from sqlalchemy.exc import DataError, IntegrityError
 from sqlalchemy.orm import lazyload
-from sqlalchemy.exc import IntegrityError, DataError
-from flask import jsonify, abort, request, current_app
+
+from dmapiclient.audit import AuditTypes
 
 from .. import main
 from ... import db, encryption
-from ...models import User, AuditEvent, Supplier, Framework, SupplierFramework, BuyerEmailDomain
-from ...utils import get_json_from_request, json_has_required_keys, \
-    json_has_matching_id, pagination_links, get_valid_page_or_1, validate_and_return_updater_request
-from ...validation import validate_user_json_or_400, validate_user_auth_json_or_400, \
-    buyer_email_address_has_approved_domain
+from ...models import AuditEvent, BuyerEmailDomain, Framework, Supplier, SupplierFramework, User
+from ...utils import (
+    get_json_from_request, json_has_required_keys, json_has_matching_id, pagination_links, get_valid_page_or_1,
+    validate_and_return_updater_request
+)
+from ...validation import (
+    buyer_email_address_has_approved_domain, validate_user_auth_json_or_400, validate_user_json_or_400
+)
+
+
+def check_supplier_role(role, supplier_id):
```

Comment:
If we're moving this round then maybe `supplier_utils.py` would be a better home for it?